client/src/interface.rs

```diff
@@ -1,90 +1,94 @@
 
-use engine::vault::{ClientId, RecordId};
+#[cfg(feature = "communication")]
+/// communication feature relevant imports are bundled here.
+mod comm {
+
+    pub use crate::utils::ResultMessage;
+    use actix::{Actor, Context};
+
+    pub use communication::{
+        actor::{
+            CommunicationActor, CommunicationActorConfig, CommunicationRequest, CommunicationResults,
+            EstablishedConnection, FirewallPermission, FirewallRule, RelayDirection, RequestDirection,
+            VariantPermission,
+        },
+        behaviour::BehaviourConfig,
+        libp2p::{Keypair, Multiaddr, PeerId},
+    };
+
+    // that's a proxy communication actor to be used
+    pub struct CommunicationActorProxy {}
+
+    impl Actor for CommunicationActorProxy {
+        type Context = Context<Self>;
+    }
+}
 
 /// The main type for the Stronghold System.  Used as the entry point for the actor model.  Contains various pieces of
 /// metadata to interpret the data in the vault and store.
 pub struct Stronghold {
-    // actor system.
-    pub system: ActorSystem,
-    // clients in the system.
-    clients: HashMap<ClientId, ActorRef<ClientMsg>>,
-    target: ActorRef<ClientMsg>,
+    registry: Addr<Registry>,
+    target: Addr<SecureClient<internals::Provider>>,
 
     #[cfg(feature = "communication")]
-    // communication actor ref
-    communication_actor: Option<ActorRef<CommunicationRequest<SHRequest, ClientMsg>>>,
+    communication_actor: Option<Addr<CommunicationActorProxy>>,
 }
 
 impl Stronghold {
-    /// Initializes a new instance of the system.  Sets up the first client actor. Accepts a [`ActorSystem`], the first
-    /// client_path: [`Vec<u8>`] and any [`StrongholdFlags`] which pertain to the first actor.
-    pub fn init_stronghold_system(system: ActorSystem, client_path: Vec<u8>, _options: Vec<StrongholdFlags>) -> Self {
-        let client_id = ClientId::load_from_path(&client_path, &client_path).expect(line_error!());
-        let id_str: String = client_id.into();
-        let mut clients = HashMap::new();
-
-        let client = system
-            .actor_of_args::<Client, _>(&id_str, client_id)
-            .expect(line_error!());
-        system
-            .actor_of_args::<InternalActor<Provider>, _>(&format!("internal-{}", id_str), client_id)
-            .expect(line_error!());
-
-        system.actor_of::<Snapshot>("snapshot").expect(line_error!());
-
-        clients.insert(client_id, client.clone());
-
-        Self {
-            system,
-            clients,
+    /// Initializes a new instance of the system.  Sets up the first client actor. Accepts an optional [`SystemRunner`],
+    /// the first client_path: `Vec<u8>` and any `StrongholdFlags` which pertain to the first actor.
+    /// - The [`SystemRunner`] is not being used directly by stronghold, but is being initialized on the first run.
+    /// - The initialization function can be made asynchronous as well, getting rid of internal explicit blocking
```

Comment:
```suggestion
    /// Initializes a new instance of the system.  Sets up the first client actor. Accepts the first client_path:
    /// `Vec<u8>` and any `StrongholdFlags` which pertain to the first actor.
```
Fix (outdated ?) doc, since there is no `SystemRunner` parameter here.